tests/README.md

```diff
@@ -1,24 +1,110 @@
-go test -kubeconfig=path/to/my/config -config=default-config.json
+export KUBEVIRT_PROVIDER=k8s-1.18
+export KUBEVIRT_NUM_NODES=2
+export KUBEVIRT_NUM_SECONDARY_NICS=2
 ```
 
-## Run them on one of the core KubeVirt providers
+- Raise a cluster based on the configuration.
+  At this stage, it will not contain kubevirt.
+  ```make cluster-up```
+- Build KubeVirt from sources.
+  ```make```
+- Deploy KubeVirt onto the cluster.
+  ```make cluster-sync```
+- Run the tests.
+  ```make functest```
+- Take down and delete the cluster.
+  ```make cluster-down```
+
+On a regular workflow of development or debugging, rebuilding KubeVirt is
+needed only when the production code is changed.
+In case only test code changes, `make functest` will rebuild the test code
+implicitly.
+
+## Existing (external) Kubevirt Cluster
+
+The tests may be executed against an already deployed (and active) Kubevirt
+cluster.
+
+The environment needs to be adjusted to allow the tests to operate agains
+such a cluster.
+The following high level steps are required to accomplish this:
+- Clone kubevirt repository on the machine that has access to the cluster.
+- Identify the deployed kubevirt version.
+- From the git repo, checkout to the commit that corrolates to the running
+kubevirt cluster.
+- Run the tests by executing the
+  [./tests/run-tests-on-external-cluster.sh](https://github.com/kubevirt/kubevirt/blob/master/tests/run-tests-on-external-cluster.sh)
+  script.
+
+### Identify the commit on which the deployed kubevirt cluster is based on
 
-There is a make target to run this with the config
-taken from hack/config.sh:
+In order to identify on which commit the current kubevirt cluster is based on,
+there is a need to query a kubevirt infra pod and then inspect its base
+container.
+
+- Lookup for the `virt-operator` pod:
+  `kubectl get pods --all-namespaces | grep "virt-operator"`
+- Use the full pod name and its namespace to inspect the container it uses:
+  `kubectl describe <virt-operator full name> -n <virt-operator namespace>`
+- Identify and record the pod image:
+  ```
+  Containers:
+  virt-operator:
+    Image:         <reference to image on the registry>
+  ```
+- Inspect the image to retrieve the commit:
+  `skopeo inspect docker://<image path>`
+  The commit ID is set on the `upstream-vcs-ref`.
+- Use git-checkout to sync the sources to the relevant commit ID.
+  `git checkout <commit id>`
+
+At this stage, the tests sources should be in sync with the running cluster
+and it is safe to setup the configuration and run the tests.
+
+### Run the tests
+
+The test environment needs to be configured to work with the running cluster
+and the test can then be build and executed against it.
+
+These steps have been automated by the `run-tests-on-external-cluster.sh`.
+
+Note that the namespace in which kubevirt components run on the cluster
+needs to be specify.
+
+`KUBEVIRTNAMESPACE="openshift-cnv" ./tests/run-tests-on-external-cluster.sh`
```

Comment:
if it's on a separate line then I think you should use 3 commas to properly markdown (```)